engine/FindGTK3.cmake

```diff
@@ -0,0 +1,611 @@
+# - FindGTK3.cmake
+# This module can find the GTK3 widget libraries and several of its other
+# optional components like gtkmm, glade, and glademm.
+#
+# NOTE: If you intend to use version checking, CMake 2.6.2 or later is
+# required.
+#
+# Specify one or more of the following components
+# as you call this find module. See example below.
+#
+#   gtk
+#   gtkmm
+#   glade
+#   glademm
+#
+# The following variables will be defined for your use
+#
+#   GTK3_FOUND - Were all of your specified components found?
+#   GTK3_INCLUDE_DIRS - All include directories
+#   GTK3_LIBRARIES - All libraries
+#
+#   GTK3_VERSION - The version of GTK3 found (x.y.z)
+#   GTK3_MAJOR_VERSION - The major version of GTK3
+#   GTK3_MINOR_VERSION - The minor version of GTK3
+#   GTK3_PATCH_VERSION - The patch version of GTK3
+#
+# Optional variables you can define prior to calling this module:
+#
+#   GTK3_DEBUG - Enables verbose debugging of the module
+#   GTK3_SKIP_MARK_AS_ADVANCED - Disable marking cache variables as advanced
+#   GTK3_ADDITIONAL_SUFFIXES - Allows defining additional directories to
+#                              search for include files
+#
+#=================
+# Example Usage:
+#
+#   Call find_package() once, here are some examples to pick from:
+#
+#   Require GTK 3.0 or later
+#       find_package(GTK3 3.0 REQUIRED gtk)
+#
+#   if(GTK3_FOUND)
+#      include_directories(${GTK3_INCLUDE_DIRS})
+#      add_executable(mygui mygui.cc)
+#      target_link_libraries(mygui ${GTK3_LIBRARIES})
+#   endif()
+#
+
+#=============================================================================
+# Copyright 2009 Kitware, Inc.
+# Copyright 2008-2009 Philip Lowman <philip@yhbt.com>
+# Copyright 2014-2018 Ettercap Development Team <info@ettercap-project.org>
+#
+# Distributed under the OSI-approved BSD License (the "License");
+# see accompanying file Copyright.txt for details.
+#
+# This software is distributed WITHOUT ANY WARRANTY; without even the
+# implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the License for more information.
+#=============================================================================
+# (To distribute this file outside of CMake, substitute the full
+#  License text for the above reference.)
+
+
+# Version 0.1 (5/13/2011)
+#   * First cut at a GTK3 version (Heavily derived from
+#     FindGTK2.cmake)
+
+# Version 0.2 (3/02/2018)
+#   * Run git diff against this file to see all changes
+#=============================================================
+# _GTK3_GET_VERSION
+# Internal function to parse the version number in gtkversion.h
+#   _OUT_major = Major version number
+#   _OUT_minor = Minor version number
+#   _OUT_micro = Micro version number
+#   _gtkversion_hdr = Header file to parse
+#=============================================================
+function(_GTK3_GET_VERSION _OUT_major _OUT_minor _OUT_micro _gtkversion_hdr)
```

Comment:
Please make all CMake directives upper case; variables can be lowercase for internal usage or uppercase for constants or exported variables.